tests/test_black.py

```diff
@@ -1512,6 +1512,73 @@ def test_parse_pyproject_toml(self) -> None:
         self.assertEqual(config["exclude"], r"\.pyi?$")
         self.assertEqual(config["include"], r"\.py?$")
 
+    def test_parse_pyproject_toml_project_metadata(self) -> None:
+        for test_toml, expected in [
+            ("only_black_pyproject.toml", ["py310"]),
+            ("only_metadata_pyproject.toml", ["py37", "py38", "py39", "py310"]),
+            ("neither_pyproject.toml", None),
+            ("both_pyproject.toml", ["py310"]),
+        ]:
+            test_toml_file = THIS_DIR / "data" / "project_metadata" / test_toml
+            config = black.parse_pyproject_toml(str(test_toml_file))
+            self.assertEqual(config.get("target_version"), expected)
+
+    @pytest.mark.filterwarnings("ignore::DeprecationWarning")
```

Comment:
We create invalid specifiers to test the code, which causes `packaging` to throw a DeprecationWarning. So in order to test that these invalid specifiers return `None` as expected, we need to ignore these warnings.

Reply:
No longer required with packaging >= 22.0